prov/rxm/src/rxm_init.c

```diff
@@ -204,10 +204,32 @@ static int rxm_init_info(void)
 	return 0;
 }
 
+static void rxm_set_data_progress(const struct fi_info *hints, struct fi_info *info)
+{
+	struct fi_info *cur;
+	int auto_progress = 0;
+
+	fi_param_get_bool(&rxm_prov, "data_progress_auto", &auto_progress);
+
+	if (!hints || !hints->domain_attr ||
+	    hints->domain_attr->data_progress == FI_PROGRESS_UNSPEC) {
+	    if (!auto_progress) {
+		for (cur = info; cur; cur = cur->next)
+			info->domain_attr->data_progress = FI_PROGRESS_MANUAL;
+	    }
+	} else {
+		for (cur = info; cur; cur = cur->next)
+			info->domain_attr->data_progress =
+				hints->domain_attr->data_progress;
+	}
+}
```

Comment:
Can you provide details on why this is needed in the commit message?  I.e. provides a work-around for apps that don't drive progress correctly...

Reply:
I've modified the commit to force auto-progress even if app has requested manual progress. Let me know if the one-line commit message is sufficient.